core/domain/exp_domain_test.py

```diff
@@ -435,6 +435,336 @@ def test_validate_exp_id(self):
             exp_domain.ExpVersionReference(0, 1)
 
 
+class StateVersionSpanTests(test_utils.GenericTestBase):
+    """Unit tests for StateVersionSpan domain objects."""
+
+    def new_state_domain_obj(self, interaction_id='TextInput'):
+        """Returns a new State domain object without putting it into storage.
+
+        Args:
+            interaction_id: str. The interaction id of the state.
+
+        Returns:
+            state_domain.State. A state with the given interaction.
+        """
+        state = state_domain.State.create_default_state('A')
+        self.set_interaction_for_state(state, interaction_id)
+        return state
+
+    def get_state_version_spans(self, name, state_v1, *subsequent_versions):
+        """Returns the state version spans for the given snapshots of a state.
+
+        Args:
+            name: str. The name of the state.
+            state_v1: state_domain.State. The first version of the state.
+            *subsequent_versions: tuple(state_domain.State). The subsequent
+                versions of the state.
+
+        Returns:
+            list(StateVersionSpan). A collection of state version spans. The
+            collection will have more than one element if a version of the state
+            is incompatible with its previous version.
+        """
+        distinct_spans = [exp_domain.StateVersionSpan(1, name, state_v1)]
+        nodiff = exp_domain.ExplorationVersionsDiff([])
+        for version, state_snapshot in enumerate(subsequent_versions, start=2):
+            span = distinct_spans[-1].extend_or_split(
+                version, name, state_snapshot, nodiff)
+            if span is not distinct_spans[-1]:
+                distinct_spans.append(span)
+        return distinct_spans
+
+    def test_set_version_directly_raises_an_error(self):
+        state = self.new_state_domain_obj()
+
+        span = exp_domain.StateVersionSpan(1, 'A', state)
+
+        with self.assertRaisesRegexp(
+            Exception, 'Only `extend_or_split` can mutate a span'):
+            span[2] = ('A', state)
+
+    def test_get_invalid_version_raises_an_error(self):
+        state = self.new_state_domain_obj()
+
+        span = exp_domain.StateVersionSpan(1, 'A', state)
+
+        with self.assertRaisesRegexp(Exception, 'does not cover version=5'):
+            _ = span[5]
+
+    def test_get_version_with_newly_constructed_span(self):
```

Comment:
I don't see any calls to get_version() in these tests which are supposed to be about get_version()? (Ditto below.)

Reply:
Done.